test-data-generator/src/main/java/com/adobe/platform/core/identity/services/datagenerator/AbstractDataGen.java

```diff
@@ -1,33 +1,34 @@
 package com.adobe.platform.core.identity.services.datagenerator;
 
-import com.adobe.platform.core.identity.services.cosmosdb.client.AsyncCosmosDbClient;
 import com.adobe.platform.core.identity.services.cosmosdb.client.CosmosDbConfig;
-import com.microsoft.azure.cosmosdb.Document;
-import com.microsoft.azure.cosmosdb.ResourceResponse;
+import com.adobe.platform.core.identity.services.cosmosdb.client.V4EncryptionCosmosDbClient;
+import com.azure.cosmos.models.CosmosItemResponse;
 import org.apache.commons.lang3.tuple.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import rx.Observable;
-import rx.schedulers.Schedulers;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
 
-import javax.print.Doc;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
 
 public abstract class AbstractDataGen implements DataGen {
     Logger logger = LoggerFactory.getLogger(AbstractDataGen.class.getSimpleName());
 
-    protected final AsyncCosmosDbClient client;
+    protected final V4EncryptionCosmosDbClient v4EncryptionCosmosDbClient;
     protected final DataGenConfig datagenConfig;
     protected final CosmosDbConfig cosmosConfig;
 
     public AbstractDataGen(DataGenConfig datagenConfig, CosmosDbConfig cosmosConfig){
-        this.client = new AsyncCosmosDbClient(cosmosConfig);
+        this.v4EncryptionCosmosDbClient = new V4EncryptionCosmosDbClient(cosmosConfig);
```

Comment:
we should have separate path for the existing code and encryption.

You should have one path for encryption and one path without and they should be separate. why are we passing everything through the encryption path now?

Reply:
We need some client in data gen which can generate data for all scenarios (V2Sync/V2Async/V4/V4 encryption). Initially we used AsyncCosmosDbClient v2 async to create the data.

Now V4 encryption is the only client which can create normal data( as it has core client as well) and encrypted data. It will save us from maintaining two clients.

Reply:
added two clients